client/driver/docker.go

```diff
@@ -884,33 +911,37 @@ func (d *DockerDriver) Periodic() (bool, time.Duration) {
 
 // createImage creates a docker image either by pulling it from a registry or by
 // loading it from the file system
-func (d *DockerDriver) createImage(driverConfig *DockerDriverConfig, client *docker.Client, taskDir *allocdir.TaskDir) error {
+//
+// Returns true if an image was downloaded and should be cleaned up.
+func (d *DockerDriver) createImage(driverConfig *DockerDriverConfig, client *docker.Client, taskDir *allocdir.TaskDir) (bool, error) {
 	image := driverConfig.ImageName
 	repo, tag := docker.ParseRepositoryTag(image)
 	if tag == "" {
 		tag = "latest"
 	}
 
-	var dockerImage *docker.Image
-	var err error
 	// We're going to check whether the image is already downloaded. If the tag
 	// is "latest", or ForcePull is set, we have to check for a new version every time so we don't
 	// bother to check and cache the id here. We'll download first, then cache.
 	if driverConfig.ForcePull {
 		d.logger.Printf("[DEBUG] driver.docker: force pull image '%s:%s' instead of inspecting local", repo, tag)
 	} else if tag != "latest" {
-		dockerImage, err = client.InspectImage(image)
+		if dockerImage, _ := client.InspectImage(image); dockerImage != nil {
+			// Image exists, nothing to do
+			return false, nil
```

Comment:
This is simply an existence check to see if we need to pull or not. We treat an error occurring as a signal to pull the image.